### PR TITLE
Allow builds to be cross compiled

### PR DESCRIPTION
**Pull request by btharper**
Adds cross compilation toolchain to dapper process when not on arm64 (currently packages are available on amd64, i386, and ppc64el from debian).

Also bumps version to 4.19 to match current branch of Raspbian.

Kernel built on amd64 tested and boots from branches 4.14.y, 4.19.y, and 5.2.y

With modifications to defconfig, dtb, firmware, and config.txt, a pi4 system can be produced on 4.19.y branch (5.2.y untested).

### CODE REVIEW

File: Dockerfile.dapper
Comment (Jason-ZW) [inline Dockerfile.dapper]: Please do not upgrade the kernel version although this is a meaningful thing, because `rancher/os` expects to upgrade the kernel after a series of stability tests.

**Review by Jason-ZW — CHANGES_REQUESTED** [review]
@btharper You also need to rebase the code. Others LGTM. After that, we will test. Thanks.

**Comment by Jason-ZW** [comment]
#1

File: Dockerfile.dapper
Comment (btharper) [reply Dockerfile.dapper]: Version bump was in service to getting things working on the Pi4, but that's probably out of scope for this PR anyway. Is leaving the minor version floating and just pinning the branch okay?

**Review by Jason-ZW — APPROVED** [review]
LGTM

**Comment by btharper** [comment]
Is the rebase just for linear history? I don't see anything conflicting across branches.

**Comment by Jason-ZW** [comment]
There's nothing more to do thank you

File: Dockerfile.dapper
Comment (Jason-ZW) [inline Dockerfile.dapper:31]: This line occurs an error. I think this is the right way `gcc-*-aarch64-linux-gnu`.

**Review by Jason-ZW — CHANGES_REQUESTED** [review]
Please modify the PR, see review comments.

File: Dockerfile.dapper
Comment (Jason-ZW) [inline Dockerfile.dapper:32]: Please formatting indentation

Comment (btharper) [reply Dockerfile.dapper:32]: Just indenting the continued command? Like this?
```
        && if [ "${DAPPER_HOST_ARCH}" != "arm64" ]; then \
                DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
                        gcc-aarch64-linux-gnu; \
                fi \
```

File: Dockerfile.dapper
Comment (btharper) [reply Dockerfile.dapper:31]: I don't receive an error while running make using this file. What host arch are you using to build? and what error are you receiving?

[gcc-aarch64-linux-gnu](https://packages.debian.org/stretch/gcc-aarch64-linux-gnu) is the general package for Debian 9. It depends on the specific package gcc-6-aarch64-linux-gn. (gcc-8 in Debian 10, gcc-9 in Debian 11).

Comment (Jason-ZW) [reply Dockerfile.dapper:31]: aws Ubuntu-18.04 aarch64 aarch64 aarch64 GNU/Linux
gcc version 7.4.0
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.2 LTS
Release:	18.04
Codename:	bionic

File: Dockerfile.dapper
Comment (Jason-ZW) [inline Dockerfile.dapper:29]: aarch64 also needs to be considered

**Review by Jason-ZW — CHANGES_REQUESTED** [review]

File: Dockerfile.dapper
Comment (Jason-ZW) [reply Dockerfile.dapper]: > Just indenting the continued command? Like this?
> 
> ```
>         && if [ "${DAPPER_HOST_ARCH}" != "arm64" ]; then \
>                 DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
>                         gcc-aarch64-linux-gnu; \
>                 fi \
> ```

```
&& if ...
          ....
      fi \
```

Comment (Jason-ZW) [reply Dockerfile.dapper:32]: my mistake that's fine

File: Dockerfile.dapper
Comment (btharper) [reply Dockerfile.dapper:29]: aarch64 and arm64 are different ways to refer to the same thing. DAPPER_HOST_ARCH is populated with the Docker server arch or the contents of GOARCH.

On my RancherOS raspberry pi I get:
`docker version -f "{{.Server.Arch}}"` gives arm64
`uname -m` gives aarch64

File: Dockerfile.dapper
Comment (btharper) [reply Dockerfile.dapper:31]: DAPPER_HOST_ARCH value is empty until the line `ARG DAPPER_HOST_ARCH` gives it a value. I've added a commit to address this. I'm currently testing on a x86_64 desktop and a RPi 3 but don't have results yet.

Comment (btharper) [reply Dockerfile.dapper:31]: Things look like they're working on both amd64 and arm64. Build on x86_64 completed without errors. RPi3 build made it past the line that was failing previously, but will not be finished quickly.

Comment (Jason-ZW) [reply Dockerfile.dapper:31]: Waiting for your RPi3 build result, after that, I will merge it.

Comment (btharper) [reply Dockerfile.dapper:31]: I'm letting it run overnight. So far, after 40 minutes, it hasn't even made it to the Dockerfile entrypoint. Kernel compile is also expected to take 3-5 hours, but I'll let you know once I have a result.

Comment (btharper) [reply Dockerfile.dapper:31]: Build finished. No errors on the last page of the scroll buffer.

**Review by Jason-ZW — APPROVED** [review]

**Pull request merged**
